tree/tree/inc/TTree.h

```diff
@@ -226,6 +229,8 @@ class TTree : public TNamed, public TAttLine, public TAttFill, public TAttMarker
    };
 
 public:
+   void InsertNameAndBranch(std::pair<std::string, TBranch *> &&kv) { fNamesToBranches.insert(kv); }
```

Comment:
This probably should not be public.  We may need to have `TBranch__SetTree` become a friend of TTree.
```suggestion
   void RegisterBranchFullName(std::pair<std::string, TBranch *> &&kv) { fNamesToBranches.insert(kv); }
```

Reply:
Done in latest iteration